src/auth/googleauth.ts

```diff
@@ -459,13 +463,74 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     return this.fromStream(readStream, options);
   }
 
+  /**
+   * Create a credentials instance using a given impersonated input options.
+   * @param json The impersonated input object.
+   * @returns JWT or UserRefresh Client with data
+   */
+  fromImpersonatedJSON(json: ImpersonatedJWTInput): Impersonated {
+    if (!json) {
+      throw new Error(
+        'Must pass in a JSON object containing an  impersonated refresh token'
+      );
+    }
+    if (json.type !== IMPERSONATED_ACCOUNT_TYPE) {
+      throw new Error(
+        `The incoming JSON object does not have the "${IMPERSONATED_ACCOUNT_TYPE}" type`
+      );
+    }
+    if (!json.source_credentials) {
+      throw new Error(
+        'The incoming JSON object does not contain a source_credentials field'
+      );
+    }
+    if (!json.service_account_impersonation_url) {
+      throw new Error(
+        'The incoming JSON object does not contain a service_account_impersonation_url field'
+      );
+    }
+
+    // Create source client for impersonation
+    const source_client = new UserRefreshClient(
+      json.source_credentials.client_id,
+      json.source_credentials.client_secret,
+      json.source_credentials.refresh_token
+    );
+
+    // Extreact service account from service_account_impersonation_url
+    const impersonation_url = json.service_account_impersonation_url;
+    const start_index = impersonation_url.lastIndexOf('/');
+    const end_index = impersonation_url.indexOf(':generateAccessToken');
+    if (start_index === -1 || end_index === -1 || start_index > end_index) {
+      throw new Error(
+        `Cannot extract target principal from ${impersonation_url}`
+      );
+    }
+
+    const target_principal = impersonation_url.substring(
+      start_index + 1,
+      end_index
+    );
```

Comment:
This works, but how about a regex for this portion? This regex captures all characters after the last `/` (as `generateAccessToken$` matches the end of the output) and labels it 'target' if it's available.

```suggestion
    const targetPrincipal = 
    /(?<target>[^/]+):generateAccessToken$/.exec(json.service_account_impersonation_url)?.groups?.target;
    
    if (!targetPrincipal) {
      throw new RangeError(
        `Cannot extract target principal from ${json.service_account_impersonation_url}`
      );
    }
```

Reply:
Will adopt to regex and explicitly test for correct targetPrincipal handling